pkg/config/types.go

```diff
@@ -59,8 +65,13 @@ type Config interface {
 	ConfigFileUsed() string
 
 	BindPFlag(key string, flag *pflag.Flag) error
+}
 
-	// Datadog specific
+// DDConfigHelpers describes the config helpers that are not provided by viper.Viper
+type DDConfigHelpers interface {
+	// GetConfigEnvVars returns a list of the non-sensitive env vars that the config supports
 	GetConfigEnvVars() []string
```

Comment:
I'd suggest (as a good practice) to always consider the name of the package to be part of exported identifiers. That way, it will be easier to come up with good names. Since this package is already called `config` it might not make sense to include the word "config" in identifiers. For example:
* prefer `EnvVars` or `GetEnvVars` (to match Viper) to `GetConfigEnvVars`
* prefer `Env` to `DDConfigHelpers` (helpers is a very generic catch-all term which will be attractive every time someone doesn't know where to put somewhere). This interface mostly addresses environment variables so it will be ok to call it `Env`
* prefer `Viper` to `ViperConfig`

At the same time (contradicting my previous comment), this could all be part of the same interface. No reason to have two now, right? The fact that that interface would also implement Viper's is just a side effect.

Reply:
that's a good point overall, thanks for commenting on this. I'll make some naming changes. To me, separating the 2 interfaces is just a way to explicitly point out what's implemented by viper and what's not, as we're in the process of investigating if/how we could replace viper, so I'll leave most of it as it is.

Reply:
getting confused by the back and forth 😅 : I'll keep only one interface, rename things, and clarify the overall intent in comments

Reply:
Sorry for creating confusion. My intention is purely to have a nice API that is easy to understand to potential newcomers. Thanks for being patient.

Reply:
no apologies needed, your feedback was very welcome :)